packages/interactivity-router/src/index.js

```diff
@@ -102,12 +134,41 @@ window.addEventListener( 'popstate', async () => {
 	}
 } );
 
-// Cache the initial page using the intially parsed vDOM.
+// Initialize the router and cache the initial page using the initial vDOM.
+// Once this code is tested and more mature, the head should be updated for region based navigation as well.
+if ( navigationMode === 'fullPage' ) {
+	// Cache the scripts. Has to be called before fetching the assets.
+	[].map.call( document.querySelectorAll( 'script[src]' ), ( script ) => {
+		headElements.set( script.getAttribute( 'src' ), {
+			tag: script,
+			text: script.textContent,
+		} );
+	} );
+	await fetchHeadAssets( document, headElements );
+}
 pages.set(
 	getPagePath( window.location ),
 	Promise.resolve( regionsToVdom( document, { vdom: initialVdom } ) )
 );
 
+// Check if the link is valid for client-side navigation.
+const isValidLink = ( ref ) =>
+	ref &&
+	ref instanceof window.HTMLAnchorElement &&
+	ref.href &&
+	( ! ref.target || ref.target === '_self' ) &&
+	ref.origin === window.location.origin;
```

Comment:
Also need to exclude `wp-admin` links. See similar logic in Customizer preview: https://github.com/WordPress/wordpress-develop/blob/f525e665b6ea6e88289d82cad5fc1dfac57db109/src/js/_enqueues/wp/customize/preview.js#L283-L323

Reply:
Also `wp-login.php`. See similar [Speculative Loading plugin exclusions](https://github.com/WordPress/performance/blob/trunk/plugins/speculation-rules/helper.php#L43).

Reply:
Also important to check for whether the link is for an anchor link for another part of the page: https://github.com/WordPress/wordpress-develop/blob/f525e665b6ea6e88289d82cad5fc1dfac57db109/src/js/_enqueues/wp/customize/preview.js#L156-L160

So this here should also check if `! ref.getAttribute('href').startsWith('#')`.

Reply:
Thanks for the feedback and the links! I added the checks for `wp-admin`, `wp-login`, `_wpnonce`, and anchors as part of [this conditional](https://github.com/WordPress/gutenberg/pull/59707/files#diff-6b0e711ee1a7c83c535664372c8a0db6b2025cb33e9263bab915dff5145e01e7R164-R167). Let me know if that's not what you had in mind.